### DISCUSSION TITLE
Team member project based permission/authorisation

### DISCUSSION THREAD

**Discussion by paul-vd** [vercel/vercel#4718] [Ideas] [answered]
It would be excellent if we could integrate some sort of team member access permission per project
Currently, you only allow the role `admin` or `member` which only affects the account and billing access.

So let's take a simple example, say you have a temporary or freelancer team member that should only have access to project X and Y, currently, if you give him access to your vercel team, you give access to all your projects within that team.

It will also add a lot of value regarding env variables because of course you can set-up permissions in Github and share your project with the temp/freelancer member, but then you have to explicitly share the development env variables as well 😞 

with such a feature this will allow a simple `vc env pull` with any authorised member to that project.

**Comment by mcsdevv** [comment, marked as answer]
Hey, Paul!

Thanks for sharing this. You're not the first person to ask for this and we're definitely looking to expand upon both the roles available, but also the level of access on a per-project basis.

As soon as there's an update or something to test, I'll let you know.
